File: GoodFoodAPI/GoodFoodAPI/Domains/Entities/NguoiDung.cs
using System;
using System.Collections.Generic;

namespace GoodFoodAPI.Domains.Entities;

public partial class NguoiDung : BaseEntity
{
    public override int Id { get; set; }

    public string? HoVaTen { get; set; }

    public string TenDangNhap { get; set; } = null!;

    public string? SoDienThoai { get; set; }

    public DateTime? NgaySinh { get; set; }

    public string MatKhau { get; set; } = null!;

    public string? Email { get; set; }

    public string? GioiTinh { get; set; }

    public int? IdHinhAnh { get; set; }

    public int LoaiNguoiDung { get; set; }

    public bool? DangNhap { get; set; }

    public byte[]? Salt { get; set; }

    public bool SetTag { get; set; }

    public virtual ICollection<BaiViet> BaiViets { get; } = new List<BaiViet>();

    public virtual ICollection<BinhLuan> BinhLuans { get; } = new List<BinhLuan>();

    public virtual ICollection<Chat> Chats { get; } = new List<Chat>();

    public virtual ICollection<Conversation> ConversationUser1s { get; } = new List<Conversation>();

    public virtual ICollection<Conversation> ConversationUser2s { get; } = new List<Conversation>();

    public virtual ICollection<DanhSachYeuThich> DanhSachYeuThiches { get; } = new List<DanhSachYeuThich>();

    public virtual ICollection<DeviceToken> DeviceTokens { get; } = new List<DeviceToken>();

    public virtual ICollection<DuLieuDeXuat> DuLieuDeXuats { get; } = new List<DuLieuDeXuat>();

    public virtual HinhAnh? IdHinhAnhNavigation { get; set; }

    public virtual ICollection<QuanAn> QuanAns { get; } = new List<QuanAn>();

    public virtual ICollection<ThongBao> ThongBaos { get; } = new List<ThongBao>();

    public virtual ICollection<BaiViet> IdBaiVietsLike { get; } = new List<BaiViet>();

    public virtual ICollection<NguoiDung> IdNguoiDungDuocTheoDois { get; } = new List<NguoiDung>();

    public virtual ICollection<NguoiDung> IdNguoiDungTheoDois { get; } = new List<NguoiDung>();
}

File: GoodFoodUI/lib/srcs/models/nguoidung.dart
class NguoiDungResponse {
  NguoiDungResponse({
    required this.id,
    required this.hoVaTen,
    required this.tenDangNhap,
    required this.soDienThoai,
    required this.ngaySinh,
    required this.matKhau,
    required this.email,
    required this.gioiTinh,
    required this.idHinhAnh,
    required this.loaiNguoiDung,
  });
  late final int id;
  late final String hoVaTen;
  late final String tenDangNhap;
  late final String soDienThoai;
  late final String ngaySinh;
  late final String matKhau;
  late final String email;
  late final String gioiTinh;
  late final int idHinhAnh;
  late final int loaiNguoiDung;

  NguoiDungResponse.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    hoVaTen = json['hoVaTen'];
    tenDangNhap = json['tenDangNhap'];
    soDienThoai = json['soDienThoai'];
    ngaySinh = json['ngaySinh'];
    matKhau = json['matKhau'];
    email = json['email'];
    gioiTinh = json['gioiTinh'];
    idHinhAnh = json['idHinhAnh'];
    loaiNguoiDung = json['loaiNguoiDung'];
  }

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['id'] = id;
    data['hoVaTen'] = hoVaTen;
    data['tenDangNhap'] = tenDangNhap;
    data['soDienThoai'] = soDienThoai;
    data['ngaySinh'] = ngaySinh;
    data['matKhau'] = matKhau;
    data['email'] = email;
    data['gioiTinh'] = gioiTinh;
    data['idHinhAnh'] = idHinhAnh;
    data['loaiNguoiDung'] = loaiNguoiDung;
    return data;
  }
}
